tensorflow/compiler/mlir/xla/transforms/hlo_legalize_to_lhlo.cc

```diff
@@ -168,15 +168,20 @@ struct HloToLHloReduceConverter
       rewriter.replaceUsesOfBlockArgument(old_arg, new_arg);
     }
     // Add an argument for the result.
-    entry_block.addArgument(
+    auto resultArg = entry_block.addArgument(
         entry_block.getArgument(original_arg_count).getType());
     // Remove the old arguments.
     for (int i = original_arg_count - 1; i >= 0; --i) {
       entry_block.eraseArgument(i);
     }
-    // Insert terminator at the end.
-    rewriter.setInsertionPointToEnd(&entry_block);
-    rewriter.create<xla_lhlo::TerminatorOp>(loc);
+
+    auto terminatorOp = entry_block.getTerminator();
+    if (terminatorOp->getNumOperands() > 0) {
+      auto operand = terminatorOp->getOperand(0);
+      OpBuilder::InsertionGuard guard(rewriter);
+      rewriter.setInsertionPoint(terminatorOp);
+      rewriter.create<xla_lhlo::CopyOp>(loc, llvm::None, operand, resultArg);
```

Comment:
Why is this only inserting a copy for one operand?

Reply:
reverted to the old behaviour, because this is the code for ReduceOp body and it works on scalars and does not really need buffer assignment. 
This was a ReturnOp inside reduce, which has always one operand..